solidity/contracts/TokenStaking.sol

```diff
@@ -193,7 +191,8 @@ contract TokenStaking is Authorizations, StakeDelegatable {
             _extraData
         );
 
-        emit Staked(_from, _operator, beneficiary, authorizer, _value);
+        emit StakeDelegated(_from, _operator);
+        emit OperatorStaked(_operator, beneficiary, authorizer, _value);
```

Comment:
Are we using two subsequent events because of the limitation of max three indexed parameters in an event?

Reply:
Yes. No more than three indexed fields and we need to be able to map between `operator/authorizer/beneficiary`. Given that those three values never change, it probably makes more sense to keep them together in the event and have the `owner` indexed in a separate one.